hooks/environment

```diff
@@ -187,6 +187,39 @@ function login() {
   fi
 }
 
+function assume_role_via_oidc_for_ecr_login() {
+  local export_credentials
+
+  echo "~~~ :buildkite::key::aws: Requesting an OIDC token for AWS from buildkite"
+
+  BUILDKITE_OIDC_TOKEN="$(buildkite-agent oidc request-token --audience sts.amazonaws.com)"
```

Comment:
We should pass `duration-seconds` through to the OIDC request, to get it baked into the JWT expiry